### PR TITLE
Fix #895: Cleaned up available search engines.

### LINKED ISSUE #895: Cleanup iOS Search Engines
Match:
https://github.com/brave/brave-browser/issues/3316

iOS Search Repo:
https://github.com/brave/Brave-iOS-SearchPlugins

Search engines to support:
- Google
- DDG
- Qwant
- StartPage
- Bing (_excluded_ from `tr`, `ru`, `th`)
- Yandex (_included_ in `tr`, `ru`, `kk`, `be`)
- Any pre-set search engines
> on iOS there is a search engine for "normal" browsing and one for "Private", so maybe up to 2 additional search engines.

- Infogalactic will be completely dropped regardless of user's set engine in either mode.

### PR DESCRIPTION
**Pull request by jhreis**
## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-ios/issues) for my issue if one did not already exist.
- [x] My patch or PR title has a standard commit message that looks like `Fix #123: This fixes the shattered coffee cup!` (or `No Bug: <message>` if no relevant ticket)
- [x] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`
- [x] New files have MPL-2.0 license header.

## Reviewer Checklist:

- [x] PR is linked to an issue via [Zenhub](https://www.zenhub.com/extension).
- [x] Issues are assigned to at least one epic.
- [x] Issues include necessary QA labels:
  - [x] `QA/(Yes|No)`
  - [x] `release-notes/(include|exclude)`
  - [x] `bug` / `enhancement`
- [x] Necessary security reviews have taken place.
- [x] Adequate test coverage exists to prevent regressions.
- [x] Adequate test plan exists for QA to validate (if applicable)

### CODE REVIEW

**Comment by jumde** [comment]
cc: @tomlowenthal

File: Client/Assets/SearchPlugins/list.json [hunk 1]
```diff
@@ -2,646 +2,32 @@
...
+      "searchDefault": "bing",
       "default": {
         "visibleDefaultEngines": [
-          "amazondotcom", 
           "bing", 
-          "google", 
-          "wikipedia-ka", 
-          "yahoo", 
-          "twitter"
+          "yandex"
```
Comment (jumde) [inline Client/Assets/SearchPlugins/list.json]: yandex is not on the list here: https://github.com/brave/brave-core/pull/1648

Comment (jhreis) [reply Client/Assets/SearchPlugins/list.json]: Coming from [here](https://bravesoftware.slack.com/archives/CELA35G5S/p1550094658036400?thread_ts=1550020220.000600&cid=CELA35G5S), but will gladly remove if needed.

Comment (srirambv) [reply Client/Assets/SearchPlugins/list.json]: Also Yandex is showing translated text for EN as well. Do we need to have this fixed or is it ok?
image

**Review by kylehickinson — COMMENTED** [review]
Isn't the SearchPlugins folder regenerated when `update_search_engines.sh` is run? Or is https://github.com/brave/Brave-iOS-SearchPlugins/blob/master/iOSFinalResult/list.json also going to being updated?

**Comment by jhreis** [comment]
@kylehickinson Yeah, I actually just deleted this script. This was based off the assumption that we wanted to continue using FF's engine support, but we have basically decided to take an entirely different approach, and will no longer update search engines this way, so completely removed script.

**Comment by kylehickinson** [comment]
@jhreis Gotcha, sounds good.

One last thing is basically around the fact that you've only removed `infogalactic.xml` but based on the new `list.json` do we still need a majority of the XML files in the `SearchPlugins` folder?

**Comment by jhreis** [comment]
@kylehickinson Yeah, this was also intentional, main thing was wanting to retain all user's default search engine selection (e.g. user sets <random search engine> as main search preference). In order to support this we need to retain all the xml files. `infogalactic.xml` was the exception and only one to be explicitly removed (users that had this selected will be auto-reset to whatever their regions default is).

**Review by kylehickinson — APPROVED** [review]

**Review by tildelowengrimm — APPROVED** [review]

**Comment by tildelowengrimm** [comment]
I don't have the subject-matter expertise to be sure, but as far as I understand what I'm reading, this looks correct.

**Pull request merged**